### PR TITLE
General improvements & API cleanup

### PR DESCRIPTION
**Pull request by filbabic**
#2721 

### 🎯 Goal

As per our discussion, we're renaming components and state to make more sense and to be easily discoverable.

I've also cleaned up the SelectedMessageOverlay API a little and fixed some UI issues. Generally, let me know if any of the names here don't make sense an we can clean them up.

My goal is for us to resolve this before we push out the "stable" compose version - we can keep this PR around, batch the changes and then merge it after the next release, so that we don't introduce breaking changes every release.

### ☑️ Checklist

- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF
name

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt [hunk 1]
```diff
@@ -60,7 +66,10 @@ public data class StreamDimens(
             attachmentsContentFileUploadWidth = 250.dp,
             threadSeparatorVerticalPadding = 8.dp,
             threadSeparatorTextVerticalPadding = 2.dp,
-            messageOverlayActionItemHeight = 40.dp
+            messageOverlayActionItemHeight = 40.dp,
+            messageOptionsMaxWidth = 200.dp,
+            messageOptionsMaxHeight = 300.dp,
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt:71]: Setting this to a smaller value to allow scroll in a popup will look weird. I think for both cases scroll should not be possible:
- The message options overlay as it is now
- Our future implementation of message options in a bottom sheet

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/SelectedMessageOverlay.kt [hunk 2]
```diff
@@ -187,24 +201,36 @@ internal fun ReactionOptionItem(
  * in the list, as those are not as important.
  *
  * @param message Message to show.
+ * @param isMine If the message is owned by the current user.
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/SelectedMessageOverlay.kt:204]: Modifier KDoc

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/state/messages/list/MessageItemState.kt [hunk 3]
```diff
@@ -1,26 +1,26 @@
-package io.getstream.chat.android.compose.state.messages.items
+package io.getstream.chat.android.compose.state.messages.list
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/state/messages/list/MessageItemState.kt:1]: 👍

**Review by bychkovdmitry — APPROVED** [review]
Great job renaming stuff and improving the UI of the message options overlay!

**Review by samiuelson — APPROVED** [review]
LGTM

**Pull request merged**